app/models/assignment.rb

```diff
@@ -155,7 +155,14 @@ def scores(questions)
 
     scores[:teams] = {}
     index = 0
-    self.teams.each do |team|
+    #search by team or user name within this assignment
+    if search_string.present?
+      filtered_teams = Team.find_by_sql(["select sub.* from (select t1.* from teams t1, teams_users tu, users u where t1.id = tu.team_id and tu.user_id = u.id and u.name like ? union select t2.* from teams t2 where t2.name like ? ) as sub where sub.parent_id = ?", "%#{search_string}", "%#{search_string}", assignment_id])
```

Comment:
Its a select statement so no potential for malicious sql commands running on the DB.